src/js/test/rpc/server.test.ts

```diff
@@ -359,6 +359,74 @@ describe("Server", function () {
       });
     });
 
+    it("should process a complete request", function () {
+      const { spyGetHandles } = createStubs(sinonInstance);
+      // transform coprocessor function
+      const uppercase = (record) => ({
+        ...record,
+        value: Buffer.from("B"),
+      });
+
+      const coprocessors = [1].map((id) =>
+        createHandle({
+          globalId: BigInt(id),
+          apply: (recordBatch) => {
+            const result = new Map();
+            const transformedRecord =
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              recordBatch.map(({ header, records }) => ({
+                header,
+                records: records.map(uppercase),
+              }));
+            result.set("result", transformedRecord);
+            return Promise.resolve(result);
+          },
+        })
+      );
+      spyGetHandles.returns(coprocessors);
+      /**
+       When we sent 1000 process requests, in the 1367 interaction the server
+       failed, because the server finishes reading a request, and when it is
+       going to read the next request, the current buffer has 14 bytes, but it
+       needs 26 minimum (Rpc header size). this commit adds the logic for
```

Comment:
This is kind of weird, talking about a "commit" in code. When this gets merged, it will become less clear which "commit" this is referencing. I would suggest rephrasing this comment.